### ISSUE TITLE
Question: Can two different laravel4-starter codebases work of the same DB?

### ISSUE THREAD

**Issue by VickG** [snipe/laravel4-starter#31]
Hello All,

I have a laravel4 starter app with 100 registered users organized in groups.

I am now launching another app that will have the exact same users and groups, but the app does something completely different from my original app.

My question is following:

Is it a good practise and is it safe to use two different applications that feed from the same database in terms of users and groups?

I want both apps to have the same user name and password. If one app resets the password, it should apply to both, thus I am thinking of using the same database.

Hoping this is possible.

Thx
Vick

**Comment by snipe** [comment]
I would strongly recommend against this. If you make changes to one system, for example, that changes the database structure of the user's table, getting those things back in sync could be a hellish nightmare.

**Comment by VickG** [comment]
There are no plans to making any changes to the user table, and even if it do I know I will need to update both systems. This is very extremely unlikely case that I am pretty sure will never happen.

Anything else you can think of?

**Comment by snipe** [comment]
Yeah, software has a way of evolving past all of our best laid plans, though. You asked if it's a good practice, and I'm saying it's not. You're welcome to do it, of course, but if I were in your shoes, I wouldn't.

**Comment by lplume** [comment]
How one would deal such a scenario? It does not look so strange in the industry to share the same users against different services. First you are right: software has a way of evolving past all of our best laid plans; usually turns out to evolving in the worst possible way. What if every application has it's own personalized table of user properties: one models per application, extending the user auth table?

**Comment by lplume** [comment]
I mean: authentication could be the same, and the way a user is retrieve could be override on the single application, joining the users properties? What you think?

**Comment by VickG** [comment]
My problem is that 2 similar applications share the exact same user base and groups, how can I base 2 apps of the same users and groups. Hoping this is possible in an easy way.

**Comment by lplume** [comment]
... anyway, just tried on my skin: it's a huge pain in the @@@

**Comment by overflowsith** [comment]
I can suggest you to manage the authentication part outside your applications, maybe exposing some API from a third app. It can be tricky to adjust your existing app, but in the future you will not have to mantain every application for every changes.
